packages/client/src/routes/story-pointing/meeting.tsx

```diff
@@ -0,0 +1,83 @@
+import { useLoaderData, useParams } from "react-router-dom";
+import { useState, useEffect } from 'react';
+import { getMeeting } from "../../models/meeting";
+import { getLoggedUser, getUsers } from "../../models/user";
+import useWebSocket from 'react-use-websocket';
+import { StoryPointingAdminControllers } from '../../components/StoryPointingAdminControllers';
+import { StoryPointingPoints } from '../../components/StoryPointingPoints';
+import { StoryPointingVotes } from '../../components/StoryPointingVotes';
+
+export async function loader({ params }) {
+  const user = await getLoggedUser();
+  const users = await getUsers();
+  const meeting = await getMeeting(params.meetingId);
+
+  return { user, users, meeting };
+}
+
+export default function Meeting() {
+  const params = useParams();
+  const { user, users: defaultUsers, meeting: defaultMeeting } = useLoaderData();
+
+  const [users, setUsers] = useState(defaultUsers);
+  const [meeting, setMeeting] = useState(defaultMeeting);
+
+  const [availablePoints, setAvailablePoints] = useState(meeting.availablePoints);
+  const [isOwner, setIsOwner] = useState(user.id === meeting.createdBy);
+  const [showVotes, setShowVotes] = useState(meeting.showVotes);
+
+  const { sendJsonMessage } = useWebSocket(`ws://localhost:3000/meeting-votes`, {
+    queryParams: {
+      userId: user.id,
+      meetingId: meeting.id,
+    },
+    onMessage: (event) => {
+      const data = JSON.parse(event.data);
+
+      if (data.users) {
+        setUsers(data.users);
+      }
+
+      if (data.meeting) {
+        setMeeting(data.meeting);
+        setAvailablePoints(data.meeting.availablePoints);
+        setShowVotes(data.meeting.showVotes);
+      }
+    },
+  });
+
+  const onVote = (vote) => {
+    setMeeting({...meeting, votes: {...meeting.votes, [user.id]: vote }});
+    sendJsonMessage({ action: 'add-vote', vote });
+  };
+
+  const onToggleResults = () => {
+    const willShowResults = !showVotes;
+    setShowVotes(willShowResults);
+    sendJsonMessage({ action: willShowResults ? 'show-votes' : 'hide-votes' });
+  };
+
+  const onReset = () => {
+    setShowVotes(false);
+    setMeeting({...meeting, votes: {}, showVotes: false });
+    sendJsonMessage({ action: 'reset-votes' });
+  };
```

Comment:
When multiple hooks work together, you usually create a custom hook to encapsulate the logic and make the component a bit more readable.
```suggestion
// pseudo-code
const useMeetingVoting = () => {
  const { user, users: defaultUsers, meeting: defaultMeeting } = useLoaderData();

  const [users, setUsers] = useState(defaultUsers);
  const [meeting, setMeeting] = useState(defaultMeeting);

  const [availablePoints, setAvailablePoints] = useState(meeting.availablePoints);
  const [isOwner, setIsOwner] = useState(user.id === meeting.createdBy);
  const [showVotes, setShowVotes] = useState(meeting.showVotes);

  const { sendJsonMessage } = useWebSocket(`ws://localhost:3000/meeting-votes`, {
    queryParams: {
      userId: user.id,
      meetingId: meeting.id,
    },
    onMessage: (event) => {
      const data = JSON.parse(event.data);

      if (data.users) {
        setUsers(data.users);
      }

      if (data.meeting) {
        setMeeting(data.meeting);
        setAvailablePoints(data.meeting.availablePoints);
        setShowVotes(data.meeting.showVotes);
      }
    },
  });

  const onVote = (vote) => {
    setMeeting({...meeting, votes: {...meeting.votes, [user.id]: vote }});
    sendJsonMessage({ action: 'add-vote', vote });
  };

  const onToggleResults = () => {
    const willShowResults = !showVotes;
    setShowVotes(willShowResults);
    sendJsonMessage({ action: willShowResults ? 'show-votes' : 'hide-votes' });
  };

  const onReset = () => {
    setShowVotes(false);
    setMeeting({...meeting, votes: {}, showVotes: false });
    sendJsonMessage({ action: 'reset-votes' });
  };
  
  return {
    // here you expose things you need
    onVote,
    onToggleResults,
    onReset,
    meeting, 
    // etc...
}
```
then use like this
```ts
const {onVote, onReset, ...etc} = useMeetingVote()
```